pkg/yamltemplate/template.go

```diff
@@ -34,6 +34,33 @@ type TemplateOpts struct {
 	ImplicitMapKeyOverrides bool
 }
 
+func IsTemplate(val interface{}) bool {
+	node, ok := val.(yamlmeta.Node)
+	if !ok {
+		return false
+	}
+
+	metaOpts := MetasOpts{IgnoreUnknown: true}
+	for _, meta := range node.GetMetas() {
+		structMeta, err := NewStructMetaFromMeta(meta, metaOpts)
```

Comment:
it seems that contract right now for NewStructMetaFromMeta is to return errors for non-valid annotations. if we load it up with `MetasOpts{}` (== `IgnoreUnknown: false`) then we can just check on the error, right?

Reply:
actually nm, this wouldnt work if there would be no comments, right...

Reply:
i guess this loop looks a little "annoying" but i dont think we can do much about it.

Reply:
> actually nm, this wouldnt work if there would be no comments, right...

I like your moxy. But, exactly.

> i guess this loop looks a little "annoying" but i dont think we can do much about it.

Yeah, it's also irritatingly identical to some linting logic we do in `DocExtractor`, but the duplication isn't tripping that threshold where the cost of the abstraction would be worth it, yet.